src/AutoMapper/TypeMap.cs

```diff
@@ -328,4 +338,39 @@ private void ApplyInheritedTypeMap(TypeMap inheritedTypeMap)
             }
         }
     }
+
+    public class TypeMap<TSource, TDestination>
+    {
+        private static readonly ConcurrentDictionary<IConfigurationProvider, TypeMap<TSource,TDestination>> Cache = new ConcurrentDictionary<IConfigurationProvider, TypeMap<TSource, TDestination>>();

```

Comment:
You can have multiple configurations making different mappers.  If both configurations map the same type map but do it 2 different ways, I'll need to know which config it came from so I can get the correct type map.

You know the questions about cloning a config and then changing a TypeMap.  Also requests to have 2 different implementations of mapping the same types to each other, based on some condition or use case.